### PR TITLE
Fix N+1 queries on the admin teachers page

### PR DESCRIPTION
**Pull request by cpjmcquillan**
### Context

This addresses some N+1 queries on the admin teachers page.

### Changes proposed in this pull request

1. Calling `count` on an ActiveRecord association **always** makes a database request. If we include `induction_periods` when retrieving the Teacher records, we can use `size` instead.

2. Similarly to #858, if we define a `current_appropriate_body` association, we can leverage ActiveRecord and `include` it without make an extra trip to the database.

### Guidance to review

**Before**
image
image

**After**
image

### CODE REVIEW

File: app/models/teacher.rb [hunk 1]
```diff
@@ -17,7 +17,10 @@ class Teacher < ApplicationRecord
   has_many :induction_periods
   has_one :first_induction_period, -> { order(started_on: :asc) }, class_name: "InductionPeriod"
   has_one :last_induction_period, -> { order(started_on: :desc) }, class_name: "InductionPeriod"
+  has_one :ongoing_induction_period, -> { ongoing }, class_name: "InductionPeriod"
```
Comment (cpjmcquillan) [inline app/models/teacher.rb:20]: There are a few places we might want to use this. Do we want to do that here or later? 💭

File: app/models/teacher.rb [hunk 2]
```diff
@@ -17,7 +17,10 @@ class Teacher < ApplicationRecord
...
+
   has_many :appropriate_bodies, through: :induction_periods
+  has_one :current_appropriate_body, through: :ongoing_induction_period, source: :appropriate_body
```
Comment (avinhurry) [inline app/models/teacher.rb:23]: Would you mind adding some tests please to confirm it returns the correct appropriate body (or `nil`) depending on whether the teacher has an ongoing induction period?

File: app/models/teacher.rb (hunk 1 above)
Comment (avinhurry) [reply app/models/teacher.rb:20]: I'm not in tomorrow but FWIW, depends how many there are - might be nicer to get this in then create follow ups?

**Review by avinhurry — COMMENTED** [review]
Nice improvement! 👍🏽 

Just 1 comment around tests in line.

File: app/models/teacher.rb (hunk 2 above)
Comment (cpjmcquillan) [reply app/models/teacher.rb:23]: Added ✅

File: app/models/teacher.rb (hunk 1 above)
Comment (cpjmcquillan) [reply app/models/teacher.rb:20]: Opened #919
